ui/src/app/views/cohort-builder/search/cohort-builder/cohort-builder.component.ts

```diff
@@ -164,55 +163,61 @@ export class CohortBuilderComponent implements OnInit, OnDestroy {
   }
 
   private updateCharts() {
-    let female = 0;
-    let male = 0;
-    let unknown = 0;
-    let aa = 0;
-    let ap = 0;
-    let c = 0;
-    let na = 0;
-    let h = 0;
-    let o = 0;
-    let u = 0;
+    /*
+     * TODO:
+     * Determine exactly how and what data is to be bundled with each Subject
+     * and whether decoding that data should take place in browser or in SQL.
+     * Until the exact encoding is finalized, this hack should generate some
+     * nice charts for a visual response to searches.
+     */
+    const genders = {
+      'Male': 0,
+      'Female': 0,
+      'Unknown': 0,
+    };
+    const races = {
+      'African American': 0,
+      'Asian/Pacific': 0,
+      'Caucasian': 0,
+      'Native American': 0,
+      'Hispanic': 0,
+      'Other': 0,
+      'Unknown': 0,
+    };
+
     this.totalSet.forEach((subject, index) => {
-      const values = subject.val.split(',');
-      if (values[1] === 'M') {
-        male++;
-      } else if (values[1] === 'F') {
-        female++;
-      } else {
-        unknown++;
+      const [uid, gender, race] = subject.val.split(',');
+      switch (gender) {
+          case '1': genders.Male++    ; break;
+          case '2': genders.Female++  ; break;
+          default : genders.Unknown++;
       }
-      if (values[2] === 'B') {
-        aa++;
-      } else if (values[2] === 'A') {
-        ap++;
-      } else if (values[2] === 'W') {
-        c++;
-      } else if (values[2] === 'I') {
-        na++;
-      } else if (values[2] === 'H') {
-        h++;
-      } else if (values[2] === 'N') {
-        o++;
-      } else {
-        u++;
+      switch (race) {
+          case '1': races.Caucasian++           ; break;
+          case '2': races['African American']++ ; break;
+          default : races.Unknown++;
```

Comment:
Does this need the other race options included in the switch statement, or is it just a placeholder? Or is this based on the `raceData` below?

Reply:
Our race codes are different than all of this (and it's possible in future these counts will all be calculated by BQ server-side). I'm assuming this is just a placeholder.

Reply:
@danqrodney that is correct, this is a placeholder.  For both gender and race (and ethnicity is its own category in Big Query) I'd like to hold off on a final version of the chart-data generation code until we know what exactly the encoding will be, whether or not we want to generate the numbers via SQL or in the front end, whether we're using 3 queries for counts or incrementally doing counts, etc etc.

@blrubenstein I don't actually know if the survey (or whatever the original data source is) has a field for Other / Unknown / I'd rather not say.  The data we have in BQ right now is pretty neatly partitioned between male and female.

Reply:
Sounds good! I just wanted to check. I assumed that it was placeholder, but it seemed worth a check just to be sure!